src/ir_Carrier.cpp

```diff
@@ -44,6 +46,15 @@ const uint16_t kCarrierAc40BitMark = 547;
 const uint16_t kCarrierAc40OneSpace = 1540;
 const uint16_t kCarrierAc40ZeroSpace = 497;
 
+const uint16_t kCarrierAc48HdrMark = 4416;
+const uint16_t kCarrierAc48BitMark = 515;
+const uint16_t kCarrierAc48HdrSpace = 4412;
+const uint16_t kCarrierAc48OneSpace = 1674;
+const uint16_t kCarrierAc48ZeroSpace = 582;
+const uint16_t kCarrierAc48SpaceGap = 5298;
```

Comment:
Yeah, this is looking more and more like `MIDEA`
These almost match perfectly with:
https://github.com/crankyoldgit/IRremoteESP8266/blob/8813010fbac0b33a9f2850f026a7b3303131cecf/src/ir_Midea.cpp#L25-L38

Reply:
> Yeah, this is looking more and more like `MIDEA`
> These almost match perfectly with:

Confirmed that my remote is using the MIDEA Protocol.   After walking through IRrmoteESP8226  step by step to figure out  why  IRrecDumpV2 could capture and translate the IR codes, but not send them.  In comparing what I coded for DECODE_CARRIER_AC48  I found that DECODE_CARRIER_AC48 was returning 

Protocol  : CARRIER_AC48
Code      : 0xA12060FFFFCE (48 Bits)

and  Media was returning,

Protocol  : MIDEA
Code      : 0xA1A060FFFF4E (48 Bits)

ir_Media is currently set in the repository with MSBfirst to false

![image](https://user-images.githubusercontent.com/50158752/82767361-560cb880-9de4-11ea-9b36-5308ef56b62e.png)
  
changing it the MSBfirst from false to true  now results in 

 Protocol  : MIDEA
Code      : 0xA12060FFFFCE (48 Bits)

In short,  this pull is no longer needed, as the the code needed in already in the repository. 

I guess all that is really needed  is  an update to ir_Media.cpp  adding 

//   Brand: Carrier,  Model: 40MKC*f (A/C) 
//   Brand: Carrier,  Model: 40MKQ*F (A/C, Heat Pump)
//   with MSBfirst in  DECODE_MIDEA set to true

Thanks for following me along on my github adventures, and for the suggestions and support.

Reply:
`MSBfirst` is actually set to true by default, the false you see is the `atleast` argument.
But there is no unittests in place that does a send and decode of this

Also, if you get a range of temp values correct (please try it out, for example moving from lowers temp and up, and make sure all values are correct, That is make sure you get MIDEA on all of them), then the decode should be correct.
And the sending have lots of tests as well, so it should be unlikely to be an issue there.

Could you give us more data on what sending works and does not?

Reply:
> 
> 
> `MSBfirst` is actually set to true by default, the false you see is the `atleast` argument.
> But there is no unittests in place that does a send and decode of this
> 
       you are correct on this,  tried to may tests today on this project,  and thought that's what I did to fix it, 

> Also, if you get a range of temp values correct (please try it out, for example moving from lowers temp and up, and make sure all values are correct, That is make sure you get MIDEA on all of them), then the decode should be correct.

 see MEDIA captured codes tab on 
https://docs.google.com/spreadsheets/d/1Et618X2dL10q0ESxuNRlSr7ePcRY1LaxBnSi8dCRKtI/edit?usp=sharing
  
theses captured codes match the state of the remote control.


> And the sending have lots of tests as well, so it should be unlikely to be an issue there.
> 
> Could you give us more data on what sending works and does not?

when using the SmartIRRepeater example,  I would get no response from the AC unit.

In testing the remote again through , to collect data for this issue,  the SmartIRRepeater example.   Half way through the temperature range, The AC unit it started to respond.

I noticed that my  serial monitor had stopped responding.  So new test.

commented out the serial.print line in  SmartIRRepeater,  and start testing again.  

all function on the remote successfully were repeated and the AC unit responded correctly.


kind of one of those duh moments.

Again thanks for the Help on this.